p/api/fever.php

```diff
@@ -0,0 +1,894 @@
+<?php
+/**
+ * Fever API for FreshRSS
+ * Version 0.1
+ * Author: Kevin Papst / https://github.com/kevinpapst
+ *
+ * Inspired by:
+ * TinyTinyRSS Fever API plugin @dasmurphy
+ * See https://github.com/dasmurphy/tinytinyrss-fever-plugin
+ */
+
+// refresh is not allowed yet, probably we find a way to support it later
+if (isset($_REQUEST["refresh"])) {
+	exit;
+}
+
+// ================================================================================================
+// BOOTSTRAP FreshRSS
+require(__DIR__ . '/../../constants.php');
+require(LIB_PATH . '/lib_rss.php');    //Includes class autoloader
+Minz_Configuration::register('system', DATA_PATH . '/config.php', FRESHRSS_PATH . '/config.default.php');
+
+// check if API is enabled globally
+FreshRSS_Context::$system_conf = Minz_Configuration::get('system');
+if (!FreshRSS_Context::$system_conf->api_enabled) {
+	Minz_Log::warning('serviceUnavailable() ' . debugInfo(), API_LOG);
+	header('HTTP/1.1 503 Service Unavailable');
+	header('Content-Type: text/plain; charset=UTF-8');
+	die('Service Unavailable!');
+}
+
+Minz_Session::init('FreshRSS');
+// ================================================================================================
+
+// this allows to overwrite the FeverAPI for special clients
+if (!function_exists('createFeverApiInstance')) {
+	function createFeverApiInstance() {
+		return new FeverAPI();
+	}
+}
+
+/**
+ * Class FeverAPI_FeedDAO for more feed functions than FreshRSS offers.
+ */
+class FeverAPI_FeedDAO extends FreshRSS_FeedDAO
+{
+	/**
+	 * @return FreshRSS_Feed
+	 */
+	public function getLastUpdatedFeed() {
+		$sql = 'SELECT id, url, name, website, `lastUpdate`, `pathEntries`, `httpAuth`, keep_history, ttl '
+			. 'FROM `' . $this->prefix . 'feed` ORDER BY `lastUpdate` LIMIT 1';
+		$stm = $this->bd->prepare($sql);
+		$stm->execute();
+
+		return current(self::daoToFeed($stm->fetchAll(PDO::FETCH_ASSOC)));
+	}
+}
+
+class FeverAPI_EntryDAO extends FreshRSS_EntryDAO
+{
+	/**
+	 * @return []
+	 */
+	public function countFever()
+	{
+		$values = [
+			'total' => 0,
+			'min' => 0,
+			'max' => 0,
+		];
+		$sql = 'SELECT COUNT(id) as `total`, MIN(id) as `min`, MAX(id) as `max` FROM `' . $this->prefix . 'entry`';
```

Comment:
It would be good to check that the clients, which we would like to support, accept 64-bit integers for ID.

Reply:
Hm, how? All I can say is that the linked clients work.

Reply:
Then it is good enough :-)